recirq/quantum_chess/circuit_transformer.py

```diff
@@ -41,7 +41,12 @@ class CircuitTransformer:
     def __init__(self):
         pass
 
-    def transform(self, circuit: cirq.Circuit) -> cirq.Circuit:
+    def __call__(
+        self,
+        circuit: cirq.AbstractCircuit,
+        *,
+        context: Optional[cirq.TransformerContext] = None,
+    ) -> cirq.Circuit:
```

Comment:
I think we can get rid of the `CircuitTransformer` interface now and simply decorate the derived classes with `@cirq.transformer` decorator -- this decorator will verify the that the derived classes adhere to `cirq.TRANSFORMER` interface and also add useful features like automated logging.